### DISCUSSION TITLE
Is it possible to make "event.all.before" callback async?

### DISCUSSION THREAD

**Discussion by Nikoms** [codeceptjs/CodeceptJS#3583] [Q&A] [answered]
Hello :)

I'm using the hooks and I'm having an async function that I need to execute in `event.all.before`. But it seems that it's never awaited and `event.suite.before` is triggered before the end of the execution.

Is there any way to handle it? Is it the same for all events?

ps: I tried to use the "recorder" but with no luck.

Thank you for your help 🙏

**Comment by adudyak** [comment, marked as answer]
Does `BeforeSuite` work for you? Put your function in it.
```
BeforeSuite(async ({ I }) => {
  await I.syncDown('testfolder');
});
```
https://codecept.io/parallel/#parallel-execution-by-workers-on-multiple-browsers

**Reply by Nikoms** [reply]
Mhh, I'll try to make it work. My goal was to use it within my event listener, but I can find a way. Thank you

**Reply by adudyak** [reply]
inside event I had smth like this in `steps_file.ts`:
```
doSomeMagic() {
  event.dispatcher.on(event.test.before, function (test) {
    console.log(test.title);
  });
}
```
then in `Scenario`:
```
I.doSomeMagic();
```

**Reply by myrepojuly** [reply]
@Nikoms I am too facing the same issue. Let me know if you are able to fix this.
